activerecord/test/cases/adapters/mysql2/connection_test.rb

```diff
@@ -55,21 +55,16 @@ def test_successful_reconnection_after_timeout_with_verify
     assert_predicate @connection, :active?
   end
 
-  def test_execute_after_disconnect
+  def test_execute_after_disconnect_reconnects
     @connection.disconnect!
 
-    error = assert_raise(ActiveRecord::ConnectionNotEstablished) do
-      @connection.execute("SELECT 1")
-    end
-    assert_kind_of Mysql2::Error, error.cause
+    assert_equal 3, @connection.select_value("SELECT 1+2")
   end
 
-  def test_quote_after_disconnect
+  def test_quote_after_disconnect_reconnects
     @connection.disconnect!
 
-    assert_raise(ActiveRecord::ConnectionNotEstablished) do
-      @connection.quote("string")
-    end
```

Comment:
Could this potentially be an undesirable change in behavior for applications?

Reply:
_Theoretically_, yes -- I believe this is the one observable difference in existing API behaviour, outside of timing.

I did consider separately tracking a concept of explicit disconnection, and having that persist until an equally-explicit reconnect... but I just can't come up with a reasonable use case we're breaking to justify the extra code / state. The obvious use for `disconnect!` would be that you e.g. want to restart the DB server, so need to break your existing session... but we still do that; we just reconnect on demand instead of leaving the adapter in a semi-permanent state of brokenness.